pkg/compiler/global_test.go

```diff
@@ -9,19 +9,35 @@ import (
 
 	"github.com/nspcc-dev/neo-go/pkg/compiler"
 	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
 	"github.com/stretchr/testify/require"
 )
 
 func TestUnusedGlobal(t *testing.T) {
-	src := `package foo
-	const (
-		_ int = iota
-		a
-	)
-	func Main() int {
-		return 1
-	}`
-	eval(t, src, big.NewInt(1))
+	t.Run("simple unused", func(t *testing.T) {
+		src := `package foo
+				const (
+					_ int = iota
+					a
+				)
+				func Main() int {
+					return 1
+				}`
+		eval(t, src, big.NewInt(1), opcode.PUSH1, opcode.RET) // Main
+	})
+	t.Run("unused with function call inside", func(t *testing.T) {
+		src := `package foo
+				var _ = f()
+				func Main() int {
+					return 1
```

Comment:
`return a+1` here and you no longer need to compare program dump (6 vs 1).

Reply:
Fixed where possible.